app/models/service/registry_api.rb

```diff
@@ -89,10 +89,37 @@ def tags_v2(image_name)
       get("/v2/#{image_name}/tags/list")['tags'].map { |tag| { 'name' => tag } }
     end
 
+    def connection_options
+      {}.merge(credentials)
+        .merge(http_proxy)
+    end
+
     def credentials
       { user: @user, password: @password }
     end
 
+    # Returns a proxy to set for connection_options
+    #
+    # If a proxy is set via ENV it will return and
+    # let Excon::Connection set it.
+    def http_proxy
+      if !env_http_proxy? && http_proxy_url
+        { proxy: http_proxy_url }
+      else
+        {}
+      end
+    end
+
+    def http_proxy_url
+      SETTINGS.fetch(:'foreman-docker', {})
+              .fetch(:http_proxy, nil)
+    end
+
+    # Checks for HTTP(S)_PROXY variables
+    def env_http_proxy?
```

Comment:
If we're using regular settings, ENV variables are probably not needed either... It's an uncommon pattern in Foreman I would say, where most settings either come from the yaml file or (better) from Administer > Settings - which allows to not restart & automatically propagates the new setting. I'd go with removing, but maybe I'm missing context, is this a ENV variable for some special reason?